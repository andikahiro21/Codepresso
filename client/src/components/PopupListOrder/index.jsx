/* eslint-disable arrow-body-style */
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Dialog } from '@mui/material';

import classes from './style.module.scss';

const PopupListOrder = ({ open, handleClose, items }) => {
  return (
    <Dialog
      className={classes.popupListOrder}
      open={open}
      fullWidth="xl"
      maxWidth="md"
      data-testid="popupListOrder"
      keepMounted
      onClose={handleClose}
      aria-describedby="alert-dialog-slide-description"
    >
      <div className={classes.cardContainer}>
        <div className={classes.cardTitle}>
          <FormattedMessage id="app_list_order_title" />
        </div>
        <div className={classes.listOrderContainer}>
          {items?.map((item) => (
            <div className={classes.listOrder}>
              <div className={classes.imgContainer}>
                <div className={classes.img}>
                  <img src={item?.menu_purchase?.image} alt="ProductImage" />
                </div>
              </div>
              <div className={classes.productDesc}>
                <div className={classes.name}>{item?.menu_purchase?.name}</div>
                <div className={classes.addons}>
                  {item?.Size && <div className={classes.size}>{item.Size.name}, </div>}
                  {item?.Bean && <div className={classes.bean}>{item.Bean.name}, </div>}
                  {item?.Sugar && <div className={classes.sugar}>{item.Sugar.name}, </div>}
                  {item?.Milk && <div className={classes.milk}>{item.Milk.name}</div>}
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </Dialog>
  );
};

PopupListOrder.propTypes = {
  handleClose: PropTypes.func,
  open: PropTypes.bool,
  items: PropTypes.object,
};

export default PopupListOrder;
