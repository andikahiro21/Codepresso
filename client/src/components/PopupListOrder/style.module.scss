@import '@styles/base';

.cardContainer {
  height: 70vh;
  padding-bottom: 1rem;
  padding-top: 1rem;
  .cardTitle {
    @include directionY;
    align-items: center;
    font-size: 1.5rem;
    font-weight: 600;
    height: 3.5rem;
    border-bottom: 1px solid rgba($color-line-grey, 0.2);
    width: 100%;
  }

  .listOrderContainer {
    width: 100%;
    height: fit-content;
    padding-bottom: 1rem;
    position: relative;
    .listOrder {
      @include directionX;
      width: 100%;
      height: 7rem;
      border-bottom: 1px solid rgba($color-line-grey, 0.2);
      .imgContainer {
        @include directionX;
        align-items: center;
        height: 100%;
        width: 40%;
        @include apply-at-xl-size {
          width: 20%;
        }
        .img {
          @include directionX;
          justify-content: center;
          align-items: center;
          background-color: $color-very-light-grey;
          height: 85%;
          width: 85%;
          border-radius: 1rem;
          margin-left: 1rem;
          img {
            width: 50%;
            @include apply-at-xl-size {
              width: 30%;
            }
          }
        }
      }
      .productDesc {
        padding-top: 0.3rem;
        padding-left: 1rem;
        position: relative;
        width: 80%;
        .name {
          font-size: 1.2rem;
          font-weight: 600;
        }
        .addons {
          @include directionX;
          gap: 0.3rem;
          font-size: 0.8rem;
          color: $color-dark-grey;
        }
      }
    }
  }
}
