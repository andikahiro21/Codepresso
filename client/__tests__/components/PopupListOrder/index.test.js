import PopupListOrder from '@components/PopupListOrder';
import { render } from '@utils/testHelper';

let wrapper;
let isOpen = true;

const mockProps = {
  open: isOpen,
  handleClose: () => {
    isOpen = false;
  },
  items: [
    {
      Bean: {
        createdAt: '2023-12-08T07:19:54.000Z',
        id: 1,
        menu_id: 1,
        name: 'Arabica',
        price: 0,
        updatedAt: '2023-12-08T07:19:54.000Z',
      },
      Milk: {
        createdAt: '2023-12-08T07:19:54.000Z',
        id: 2,
        menu_id: 1,
        name: 'Oat Milk',
        price: 5000,
        updatedAt: '2023-12-08T07:19:54.000Z',
      },
      Size: {
        createdAt: '2023-12-08T07:19:54.000Z',
        id: 2,
        menu_id: 1,
        name: 'Large',
        price: 7000,
        updatedAt: '2023-12-08T07:19:54.000Z',
      },
      Sugar: {
        createdAt: '2023-12-08T07:19:54.000Z',
        id: 2,
        menu_id: 1,
        name: 'Normal Sugar',
        price: 0,
        updatedAt: '2023-12-08T07:19:54.000Z',
      },
      bean_id: 1,
      createdAt: '2023-12-10T14:41:16.000Z',
      id: 55,
      menu_id: 1,
      menu_purchase: {
        category_id: 2,
        createdAt: '2023-12-07T14:58:23.000Z',
        description:
          'Caffe latte, often referred to as simply "latte," is a creamy and comforting coffee beverage made by combining a shot of espresso with steamed milk. It offers a perfect balance between the rich flavors of espresso and the smoothness of milk, making it a popular choice among coffee enthusiasts worldwide.',
        id: 1,
        image: 'https://res.cloudinary.com/dem9rzjbs/image/upload/v1701961102/images/bysgz0k8huekxhmogoaj.png',
        name: 'Caffe Latte',
        price: 24000,
        qty: 1,
        type: 'Beverage',
        updatedAt: '2023-12-07T14:58:23.000Z',
      },
      milk_id: 2,
      price: 108000,
      purchase_group_id: 20,
      qty: 3,
      size_id: 2,
      sugar_id: 2,
      updatedAt: '2023-12-10T14:41:16.000Z',
      user_id: 2,
    },
  ],
};

beforeEach(() => {
  wrapper = render(
    <PopupListOrder open={mockProps.open} handleClose={mockProps.handleClose} items={mockProps.items} />
  );
});

describe('Popup List Order Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('popupListOrder')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
