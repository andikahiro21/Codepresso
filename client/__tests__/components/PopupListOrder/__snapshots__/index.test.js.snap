// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup List Order Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root popupListOrder MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      data-testid="popupListOrder"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-describedby="alert-dialog-slide-description"
          aria-labelledby=":r1:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth css-rnmm7m-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="cardContainer"
          >
            <div
              class="cardTitle"
            >
              Daftar Pesanan
            </div>
            <div
              class="listOrderContainer"
            >
              <div
                class="listOrder"
              >
                <div
                  class="imgContainer"
                >
                  <div
                    class="img"
                  >
                    <img
                      alt="ProductImage"
                      src="https://res.cloudinary.com/dem9rzjbs/image/upload/v1701961102/images/bysgz0k8huekxhmogoaj.png"
                    />
                  </div>
                </div>
                <div
                  class="productDesc"
                >
                  <div
                    class="name"
                  >
                    Caffe Latte
                  </div>
                  <div
                    class="addons"
                  >
                    <div
                      class="size"
                    >
                      Large
                      , 
                    </div>
                    <div
                      class="bean"
                    >
                      Arabica
                      , 
                    </div>
                    <div
                      class="sugar"
                    >
                      Normal Sugar
                      , 
                    </div>
                    <div
                      class="milk"
                    >
                      Oat Milk
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
